File: src/main/java/ch/andreskonrad/torenta/directory/dto/FileHierarchyDto.java
package ch.andreskonrad.torenta.directory.dto;

public class FileHierarchyDto {

    private final MoviesRootDirectoryDto moviesRootDirectoryDto;
    private final SeriesRootDirectoryDto seriesRootDirectoryDto;
    private final String absolutePath;

    public FileHierarchyDto(MoviesRootDirectoryDto moviesRootDirectoryDto, SeriesRootDirectoryDto seriesRootDirectoryDto, String absolutePath) {
        this.moviesRootDirectoryDto = moviesRootDirectoryDto;
        this.seriesRootDirectoryDto = seriesRootDirectoryDto;
        this.absolutePath = absolutePath;
    }

    public MoviesRootDirectoryDto getMoviesRootDirectoryDto() {
        return moviesRootDirectoryDto;
    }

    public SeriesRootDirectoryDto getSeriesRootDirectoryDto() {
        return seriesRootDirectoryDto;
    }

    public String getAbsolutePath() {
        return absolutePath;
    }
}

File: frontend/src/app/shared/dto/directory/FileHierarchyDto.ts
import {MoviesRootDirectoryDto} from "./MoviesRootDirectoryDto";
import {SeriesRootDirectoryDto} from "./SeriesRootDirectoryDto";

export interface FileHierarchyDto {

  moviesRootDirectoryDto: MoviesRootDirectoryDto;
  seriesRootDirectoryDto: SeriesRootDirectoryDto;
  absolutePath: string;
}
